src/document.c

```diff
@@ -619,8 +618,8 @@ FIELD_PREPROCESSOR(vectorPreprocessor) {
   }
   if (fdata->vecLen != fs->vectorOpts.expBlobSize) {
 
-    QueryError_SetErrorFmt(status, QUERY_EBADATTR,
-                           "Could not add vector with blob size %zu (expected size %zu)", fdata->vecLen,
+    QueryError_SetWithUserDataFmt(status, QUERY_EBADATTR,
```

Comment:
Is Blob size considered user data? Protecting user data in this context is about privacy and hiding sensitive data. Do we regard the length of some data as sensitive?